compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -611,13 +609,32 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
       }
     }
 
+    // replace uninstantiated type vars with WildcardType, check tests/patmat/3333.scala
+    val instUndetMap = new TypeMap {
+      def apply(t: Type): Type = t match {
+        case tvar: TypeVar if !tvar.isInstantiated => WildcardType(tvar.origin.underlying.bounds)
+        case _ => mapOver(t)
+      }
+    }
+
+    val force = new ForceDegree.Value(
+      tvar => !(ctx.typerState.constraint.entry(tvar.origin) eq tvar.origin.underlying),
+      minimizeAll = false
+    )
+
     val tvars = tp1.typeParams.map { tparam => newTypeVar(tparam.paramInfo.bounds) }
     val protoTp1 = thisTypeMap(tp1.appliedTo(tvars))
 
-    if (protoTp1 <:< tp2 && isFullyDefined(protoTp1, ForceDegree.noBottom)) protoTp1
+    if (protoTp1 <:< tp2) {
+      isFullyDefined(protoTp1, force)
+      instUndetMap(protoTp1)
```

Comment:
If I understand correctly how `isFullyDefined` behaves, then if it returns `true`, all TVars in `protoTp1` should be already instantiated and `instUndetMap` will be a no-op. Would it make sense to only use `instUndetMap` if `isFullyDefined` returned `false`?

Reply:
Yes, good catch, I've updated the code, thanks a lot.